io-sim/src/Control/Monad/IOSim/Types.hs

```diff
@@ -771,10 +775,12 @@ data SimEventType
                        (Maybe Effect)    -- effect performed (only for `IOSimPOR`)
   | EventTxWakeup      [Labelled TVarId] -- changed vars causing retry
 
-  | EventTimerCreated   TimeoutId TVarId Time
-  | EventTimerUpdated   TimeoutId        Time
-  | EventTimerCancelled TimeoutId
-  | EventTimerExpired   TimeoutId
+  | EventThreadDelay          TimeoutId        Time
+  | EventRegisterDelayCreated TimeoutId TVarId Time
```

Comment:
But note that they don't have their own expiry/end events. They use the same `EventTimerExpired` event. Arguably they should use their own. At least, `EventThreadDelay` should use its own, and would not need any `TimeoutId` since it happens only in the context of a single thread. The `EventRegisterDelayCreated` does indeed still need the `TimeoutId` since it's based on an independent `TVar` rather than being tied to a single thread.

Reply:
Done!